securedrop_client/app.py

```diff
@@ -150,7 +156,14 @@ def start_app(args, qt_args) -> None:
 
 
 def run() -> None:
+    config_file = "/etc/securdrop-client/client.ini"
     args, qt_args = arg_parser().parse_known_args()
+    if args.sdc_home == DEFAULT_SDC_HOME and \
+            os.path.exists(config_file):  # pragma: no cover
+        config = configparser.ConfigParser()
+        config.read(config_file)
+        args.sdc_home = config["client"]["homedir"]
+        args.proxy = config["client"]["use_securedrop_proxy"]
     # reinsert the program's name
     qt_args.insert(0, 'securedrop-client')
```

Comment:
More generally, I think that the CLI parsing should all happen outside the app, and then we pass those as args to `start_app` (or `run_app`, whatever the name was). In side there, we handle the logic of how to include values from the configs. I think reusing the `Namespace` from `argparse` as the "master config" seems wrong. We might want to have a `Config` class that does all this logic.

Reply:
I am going to leave this for a future PR.